o3de/o3de
# Action Manager | Fix issues that resulted in prefab focus and entity selection being lost when leaving component mode via Esc

## What does this PR do?
Introduces a few fixes to the Action Manager:

- Restored an old fix that prevents Shortcuts from being auto-repeated when the key sequence is held. This is a default in Qt, but given the peculiarities of O3DE it usually results in the Editor getting overloaded and is rarely intentional on the users' part. UX consensus was agreed when this fix was applied to the legacy Action Manager years ago, and is just being ported over;
- Fixed issue in the Action Context widget event filter where actions bound to a shortcut were identified and triggered in a single loop. That could result in undefined behavior since triggering an action could change the enabled state of other actions while the loop was still happening, making it so that which actions are executed depended on the order they were registered;
- Migrated the `Esc` actions for the entity selection and prefab focus mode to use Actions in the new Action Manager.
- Removed a workaround to deliver the `Esc` shortcut event via the Viewport that was necessary with the legacy ShortcutDispatcher but was now causing the shortcut to be triggered twice when sent via the viewport.

The current setup of the `Esc` keypress in the Editor partially still bypasses the Action Manager. This is fine since the Action Manager actions are unreachable in Game Mode, and the custom code enforces that `Esc` can be pressed to leave it. The other code path is now only used in Entity Pick mode, and even after migrating fully to the new Action Manager I would be ok with leaving it as is since Ctrl interactions and shortcuts are not (and probably should not) be migrated to the Action Manager.

## How was this PR tested?

Manual testing.

Before
![RetainSelectionOnComponentModeEsc_before](https://user-images.githubusercontent.com/82231674/215570907-e91be657-f2e8-4445-a4a3-5370cded8cb8.gif)

After
![RetainSelectionOnComponentModeEsc](https://user-images.githubusercontent.com/82231674/215570925-d0f5b29c-3fbd-4b75-83b9-6af3bb940d8e.gif)

Test steps:

(Need to have a level loaded with a nested prefab containing at least one entity with a component that has a component mode).

- Focus on the nested prefab
- Select the entity inside the nested prefab
- Switch to a Component Mode
- Press `Esc`

Expected result: Component Mode is closed, but the nested prefab is still in focus and the the entity selection persists.

Actual result (before this change): Component Mode, prefab focus and selection are all cleared.
